src/lib/y2partitioner/actions/delete_partition.rb

```diff
@@ -29,13 +29,94 @@ module Actions
     #
     # @see DeleteDevice
     class DeletePartition < DeleteDevice
+      # Confirmation message before performing the delete action
+      def confirm
+        if used_by_lvm?
+          confirm_for_used_by_lvm
+        elsif used_by_md_raid?
+          confirm_for_used_by_md_raid
+        else
+          super
+        end
+      end
+
       # Deletes the indicated partition (see {DeleteDevice#device})
       def delete
         log.info "deleting partition #{device}"
         disk_device = device.partitionable
         disk_device.partition_table.delete_partition(device)
         UIState.instance.select_row(disk_device)
       end
+
+    private
+
+      # Checks whether the partition is used as physical volume
+      #
+      # @return [Boolean] true if partition belongs to a volume group; false otherwise
+      def used_by_lvm?
```

Comment:
There is a Proposal::PartitionKiller#lvm_vg? that is too similar to this. Time to have a method `Partition#lvm_vg` (or `BlkDevice#lvm_vg`)?

Reply:
Or maybe a more explicit name to make obvious that the VG is a descendant of the block device and not the other way around.